lua/entities/gmod_wire_expression2/core/custom/stcontrol.lua

```diff
@@ -0,0 +1,1094 @@
+--[[ ******************************************************************************
+ My custom state LQ-PID control type handling process variables
+****************************************************************************** ]]--
+
+local pairs        = pairs
+local tostring     = tostring
+local tonumber     = tonumber
+local getmetatable = getmetatable
+local CreateConVar = CreateConVar
+local bitBor       = bit.bor
+local mathAbs      = math.abs
+local mathModf     = math.modf
+local tableConcat  = table.concat
+local tableInsert  = table.insert
+local tableRemove  = table.remove
+local getTime      = CurTime -- Using this as time benchmarking supporting game pause
+
+-- Register the type up here before the extension registration so that the state control still works
+registerType("stcontrol", "xsc", nil,
+  nil,
+  nil,
+  function(retval)
+    if(retval == nil) then return end
+    if(not istable(retval)) then error("Return value is neither nil nor a table, but a "..type(retval).."!",0) end
+  end,
+  function(v)
+    return (not istable(v))
+  end
+)
+
+--[[ ****************************************************************************** ]]
+
+E2Lib.RegisterExtension("stcontrol", true, "Lets E2 chips have dedicated state control objects")
+
+-- Client and server have independent value
+local gnIndependentUsed = bitBor(FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_PRINTABLEONLY)
+-- Server tells the client what value to use
+local gnServerControled = bitBor(FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_PRINTABLEONLY, FCVAR_REPLICATED)
+
+local gtComponent = {"P", "I", "D"} -- The names of each term. This is used for indexing and checking
+local gsFormatPID = "(%s%s%s)" -- The general type format for the control power setup
+local gtMissName  = {"Xx", "X", "Nr"} -- This is a place holder for missing/default type
+local gtStoreOOP  = {} -- Store state controllers here linked to the entity of the E2
+local gsVarPrefx  = "wire_expression2_stcontrol" -- This is used for variable prefix
+local gtStringMT  = getmetatable(gsVarPrefx) -- Store the string metatable
+local varMaxTotal = CreateConVar(gsVarPrefx.."_max" , 20, gnServerControled, "StControl items maximum count")
+local varEnStatus = CreateConVar(gsVarPrefx.."_enst",  0, gnIndependentUsed, "Enables status output messages")
+local gsDefPrint  = "TALK" -- Default print location
+local gtPrintName = {} -- Contains the print location specification
+      gtPrintName["NOTIFY" ] = HUD_PRINTNOTIFY
+      gtPrintName["CONSOLE"] = HUD_PRINTCONSOLE
+      gtPrintName["TALK"   ] = HUD_PRINTTALK
+      gtPrintName["CENTER" ] = HUD_PRINTCENTER
+
+local function isEntity(vE)
+  return (vE and vE:IsValid())
+end
+
+local function isHere(vV)
+  return (vV ~= nil)
+end
+
+local function isString(vS)
+  return (getmetatable(vS) == gtStringMT)
```

Comment:
isstring function already exists. Use that instead as it's much more efficient than this.